src/transport.ts

```diff
@@ -1,65 +1,67 @@
+import { WebRTCDialOptions } from './options'
+import { WebRTCStream } from './stream'
+import { noise as Noise } from '@chainsafe/libp2p-noise'
+import { Connection } from '@libp2p/interface-connection'
+import type { PeerId } from '@libp2p/interface-peer-id'
+import { CreateListenerOptions, Listener, symbol, Transport } from '@libp2p/interface-transport'
+import { logger } from '@libp2p/logger'
+import { Multiaddr } from '@multiformats/multiaddr'
+import { v4 as genUuid } from 'uuid'
+import defer from 'p-defer'
+import { fromString as uint8arrayFromString } from 'uint8arrays/from-string'
+import { concat } from 'uint8arrays/concat'
+import * as multihashes from 'multihashes'
+import { dataChannelError, inappropriateMultiaddr, unimplemented, invalidArgument } from './error'
+import { WebRTCMultiaddrConnection } from './maconn'
+import { DataChannelMuxerFactory } from './muxer'
+
+const log = logger('libp2p:webrtc:transport')
+const HANDSHAKE_TIMEOUT_MS = 10000
+const WEBRTC_CODE: number = 280
+const CERTHASH_CODE: number = 466
 
 export interface WebRTCTransportComponents {
   peerId: PeerId
 }
 
 export class WebRTCTransport implements Transport {
-  private components: WebRTCTransportComponents
+  private readonly components: WebRTCTransportComponents
 
-  constructor(components: WebRTCTransportComponents) {
+  constructor (components: WebRTCTransportComponents) {
     this.components = components
   }
 
-  async dial(ma: Multiaddr, options: WebRTCDialOptions): Promise<Connection> {
-    const rawConn = await this._connect(ma, options);
-    log(`dialing address - ${ma}`);
-    return rawConn;
+  async dial (ma: Multiaddr, options: WebRTCDialOptions): Promise<Connection> {
+    const rawConn = await this._connect(ma, options)
+    log(`dialing address - ${ma.toString()}`)
+    return rawConn
   }
 
-  createListener(options: CreateListenerOptions): Listener {
-    throw unimplemented('WebRTCTransport.createListener');
+  createListener (options: CreateListenerOptions): Listener {
+    throw unimplemented('WebRTCTransport.createListener')
   }
 
-  filter(multiaddrs: Multiaddr[]): Multiaddr[] {
-    return multiaddrs.filter(validMa);
+  // Filter out invalid multiaddrs
+  filter (multiaddrs: Multiaddr[]): Multiaddr[] {
+    return multiaddrs.filter(validMa)
   }
 
-  get [Symbol.toStringTag](): string {
-    return '@libp2p/webrtc';
+  // Implement toString() for WebRTCTransport
+  get [Symbol.toStringTag] (): string {
+    return '@libp2p/webrtc'
   }
 
-  get [symbol](): true {
-    return true;
+  get [symbol] (): true {
+    return true
   }
 
-  async _connect(ma: Multiaddr, options: WebRTCDialOptions): Promise<Connection> {
-    const rps = ma.getPeerId();
-    if (!rps) {
-      throw inappropriateMultiaddr("we need to have the remote's PeerId");
+  // Connect to a peer
+  async _connect (ma: Multiaddr, options: WebRTCDialOptions): Promise<Connection> {
+    const rps = ma.getPeerId()
+    if (rps === null) {
```

Comment:
```suggestion
    if (!rps) {
```
?

Reply:
In general, the linter didn't like these type coercions and wanted explicit types.  When I turn this on, I get:
```
Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly. [Error/@typescript-eslint/strict-boolean-expressions]
```